runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangePlugin.java

```diff
@@ -679,6 +682,54 @@ public void onScriptCallbackEvent(ScriptCallbackEvent event)
 		stringStack[stringStackSize - 1] += titleBuilder.toString();
 	}
 
+	private void setLimitReset(int itemId)
+	{
+		String lastLimit = getLimitReset(itemId);
+		if (!lastLimit.equals("None"))
+		{
+			return; // do not update if previous limit has not expired
+		}
+		String itemIdStr = Integer.toString(itemId);
+		Instant currentDateTime = Instant.now();
+		configManager.setConfiguration("grandexchange",
+			client.getUsername().toLowerCase() + itemIdStr, currentDateTime);
+	}
+
+	private String getLimitReset(int itemId)
```

Comment:
This should be split into two methods, one for the UI specific stuff ("None" + the time formatter) and one for the config stuff (reading the key, checking the time difference)`

Reply:
i wasnt able to figure out how to split the functionality, they seem pretty well coupled. The method just returns a string of the remaining time to limit reset (read key and check diff, then return diff in string format)

Reply:
What about https://github.com/Adam-/runelite/commit/74206c398d88aa6d3f4e7e9a791a9807356dfbb6?

Reply:
@Adam- Sure I guess, as long as the functionality is in. What do I need to do?

Reply:
> @Adam- Sure I guess, as long as the functionality is in. What do I need to do?

email him, he's busy marking my posts as spam but won't reply to you.

I'm not sure what I've done?

Reply:
Can you check the functionality is what you expect?

Reply:
Yes it is as expected. The code is cleaner because you made use of Instant and Duration. Something for me to learn (don't use strings all the time!).